Polynomials.Core/Polynomial.cs

```diff
@@ -0,0 +1,198 @@
+﻿using System;
+using System.Text;
+
+namespace Polynomials.Core
+{
+    public class Polynomial : IEquatable<Polynomial>, ICloneable
+    {
+        private readonly float[] _coefficients;
+
+        public float this[int i]
+        {
+            get => _coefficients[i];
+            set => _coefficients[i] = value;
+        }
+
+        public Polynomial(float[] coefficients)
+        {
+            if (coefficients.Length > 100)
+            {
+                throw new Exception("Max power must be less or equal 100");
```

Comment:
I don't know why I haven't noticed that before, but throwing just Exception is generally a bad practice. Please look at the existing exception types in .NET to find the one which suits this situation.